### ISSUE TITLE
Define how to handle console report

### ISSUE THREAD

**Issue by camilamaia** [scanapi/scanapi#132]
## Description

- How can we know if user wants to print the custom template in the terminal or have it in a file.
- template and report options can be sent together?

**Comment by camilamaia** [comment]
I came up with this structure:

Every time ScanAPI runs, it generates:
- Console Logs (Writing doc, making request x, making request y...)
- A console report
- An output file containing the html report (usually `scanapi-report.html`)

Besides the extra flags:

- `--output-path` to rename the output file. To use another path instead of the default `scanapi-report.html`

- `--reporter` with the options `html` and `markdown`. This is related only to the **default output** reports available. If html is selected, it will generate the **default html report**. If markdown is selected, it will generate **the default markdown report**.

- `--console-template` for the users that want to personalize the console report.

- `--template` for the users that want to personalize the **output** report. Flag `--reporter` would be **ignored** in this case. The default output file name will keep being `scanapi-report.html`. If the user wants to change it, he/she should send the new file path using the `--reporter` flag.

What do you think @barbosa and @gillianomenezes?

**Comment by barbosa** [comment]
I like the shape this conversation is turning into.

I believe the main issue here is that we were mixing 2 different concepts into 1: debugging (console printing) and execution (report file generation).

For the sake of simplicity, on top of what you said, I would suggest:

1. Remove the `console` report and its template as an option. This would allow us to remove many pieces of code where we check whether or not this is a console reporter to do X instead of Y. Also, I don't think that always generating a html/md file is something bad enough for the developer experience, since the execution time is almost the same as printing in console. Some people may argue that this would accidently override the previous report but idk if this is a big issue.
2. Not have a `--console-template` option and leave this for the future if we realize there's a real need. This could also be a source of confusion since we'd have 2 different ways to pass templates as arguments

About `--no-output` and `--no-console-report`, I have mixed feelings. It seems to me they are flags intended to reduce complexity. But maybe we don't need the original complexity at all if debugging and execution are two isolated responsibilities.

Users will want to debug (true/false) and generate reports (true/false). This brings us to 4 combinations of the 2 variables:
- debug=False, report=False (non sense)
- debug=False, report=True (big majority of the cases users want)
- debug=True, report=False (I don't see users wanting to do this often)
- debug=True, report=True (this, maybe, and it could be resolved with a `--verbose` option perhaps?)

**Comment by camilamaia** [comment]
@barbosa I strongly agree with you. My solution was thought in a way to make things more flexible. But we don't know if this is indeed a need. So let me just recap with you:

- We will remove the console report at all

The related flags will be:

- `--output-path` to rename the output file. To use another path instead of the default `scanapi-report.html`

- `--reporter` with the options `html` and `markdown`. If html is selected, it will generate the **[default html report](https://github.com/scanapi/scanapi/blob/master/scanapi/templates/html.jinja)**. If markdown is selected, it will generate **[the default markdown report](https://github.com/scanapi/scanapi/blob/master/scanapi/templates/markdown.jinja)**.

- `--template` for the users that want to customize their reports. The default output file name will keep being `scanapi-report.html`. If the user wants to change it, he/she should send the new file path using the --output-path flag.

Is that right, @barbosa ?

**Comment by barbosa** [comment]
@camilamaia sounds like a plan. If we want to bring console printing back, I'd do that by having a `--verbose` option.

**Comment by camilamaia** [comment]
Perfect, sounds good to me!
